### ISSUE TITLE
azurerm_data_factory_pipeline doesn't support Fail Activity

### ISSUE THREAD

**Issue by vasilecelac** [hashicorp/terraform-provider-azurerm#14367]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_2.87.0`

### Terraform Configuration Files
```hcl
resource "azurerm_data_factory_pipeline" "failPipeline" {
name = "Test fail"
resource_group_name = azurerm_resource_group.resource_group.name
data_factory_name = azurerm_data_factory.data_factory.name
variables = {}
depends_on = []
activities_json = <<JSON
[
{
"name": "Test Fail",
"type": "Fail",
"dependsOn": [],
"userProperties": [],
"typeProperties": {
"message": "Failed testing testing",
"errorCode": "500"
}
}
]
JSON
}
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by favoretti** [comment]
Hi there and thank you for reporting this. Would you mind providing sample terraform code?

**Comment by vasilecelac** [comment]
> 
resource "azurerm_data_factory_pipeline" "failPipeline" {
name                = "Test fail"
resource_group_name = azurerm_resource_group.resource_group.name
 data_factory_name   = azurerm_data_factory.data_factory.name
variables           = {}
 depends_on          = []
 activities_json     = <<JSON
[
{
   "name": "Test Fail",
    "type": "Fail",
    "dependsOn": [],
    "userProperties": [],
    "typeProperties": {
          "message": "Failed testing testing",
          "errorCode": "500"
     }
}
]
JSON
}

**Comment by favoretti** [comment]
@vasilecelac thank you for providing the config. What exactly isn't supported? Is there a particular error you're getting? From HCL perspective we can't do anything here really, as it's just an activity JSON value for an already supported parameter field.

**Comment by vasilecelac** [comment]
> @vasilecelac thank you for providing the config. What exactly isn't supported? Is there a particular error you're getting? From HCL perspective we can't do anything here really, as it's just an activity JSON value for an already supported parameter field.

@favoretti terraform doesn't understand Activity type: "Fail". Activity can't be edited.

**Comment by favoretti** [comment]
Unfortunately current SDK doesn't know of type `Fail`: https://github.com/Azure/azure-sdk-for-go/blob/v59.4.0/services/datafactory/mgmt/2018-06-01/datafactory/enums.go#L1702

So this should be logged as an issue against the SDK before we can do anything about it.

**Comment by MDel-BI** [comment]
i have the same problem deploying Fail activities in Azure Data Factory. Is there already a fix or workaround?

**Comment by rcskosir** [comment]
This appears to have been added to the SDK as can be seen here: https://github.com/Azure/azure-sdk-for-go/blob/2187aca40bbf6931ffac666977e5d29128d82585/services/datafactory/mgmt/2018-06-01/datafactory/enums.go#L1841.
This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
